src/Backend/Modules/Pages/Domain/ModuleExtra/ModuleExtraRepository.php

```diff
@@ -127,4 +128,31 @@ public function getWidgetId(string $module, string $action): ?int
 
         return $moduleExtra->getId();
     }
+
+    public function updateWidgetDataByModuleAndSequence(string $module, string $sequence, $data): void
```

Comment:
doesn't sequence need to be an int?
Also, I would make it more generic and add the type as a parameter